src/lib.rs

```diff
@@ -2,26 +2,48 @@ pub mod env;
 pub mod stats;
 pub mod store;
 
-use crate::stats::CpuUsageValue;
+use crate::stats::{
+    CgroupVersion, CpuUsageValue, detect_cgroup_version, get_cgroup_v1_mount_points,
+    get_cgroup_v2_mount_point,
+};
 use crate::store::StatsEntry;
 use std::thread;
 use tracing::{debug, error};
 
 pub fn run_acolyte() {
     let stat_interval = env::get_stat_interval();
     loop {
+        let cgroup_detect = detect_cgroup_version("/proc/self/cgroup").ok();
+        let v2_mount_point = match cgroup_detect {
+            Some(CgroupVersion::V2) => get_cgroup_v2_mount_point("/proc/mounts").ok(),
+            _ => None,
+        };
+        let v1_mount_points = match cgroup_detect {
+            Some(CgroupVersion::V1) => get_cgroup_v1_mount_points("/proc/mounts").ok(),
+            _ => None,
+        };
+
         let mut stats_entry = StatsEntry::new();
 
-        let maybe_num_cpus = stats::get_num_cpus();
+        let maybe_num_cpus = stats::get_num_cpus(
+            cgroup_detect.clone(),
+            v2_mount_point.clone(),
+            v1_mount_points.clone(),
+        );
         if let Some(num_cpus) = maybe_num_cpus {
             stats_entry.num_cpus = Some(num_cpus);
         }
 
-        if let Some(cpu_usage) = stats::get_cpu_usage() {
+        if let Some(cpu_usage) = stats::get_cpu_usage(
+            cgroup_detect.clone(),
+            v2_mount_point.clone(),
+            v1_mount_points.clone(),
+        ) {
```

Comment:
So much cloning going on in these... But that can be tackled in a follow-up.

Reply:
they were `&Option<_>` references first but that felt an overkill for cloning of 4 short strings + enum 3 times per loop 😅

could maybe look into using Path instead of PathBuf though 🤔

Reply:
quickly tried to turn them to references but the underlying structures assume to take ownership for themselves so maybe I'll tackle this later as might need a bit more changes

meaning, turning them to references atm would only move the `.clone` one layer down 😅